manifests/agent.pp

```diff
@@ -80,9 +101,14 @@
       "${backend_protocol}://${backend}"
     }
   }
-  $default_config = {
-    'backend-url' => $backend_urls,
-  }
+  $default_config = delete_undef_values({
```

Comment:
Will this need to change as new types are added? If so, this would be good to call out in the DESIGN doc or similar.

Reply:
No, it won't. This is not impacted by types.